### DISCUSSION TITLE
regex search capability within Jinja template

### DISCUSSION THREAD

**Discussion by rajeshbadveti** [pallets/jinja#1785] [Q&A]
Hello Team, 
I am looking for help to perform regex operation within jinja template.

For example, below are the commands that will be included in template:

_sms all status_:
```
SMS server
Enabled: true
Status: running PID:29157 for 1377474s

SMS server rate
Enabled: true
Status: running PID:27468 for 1377466s

SMS server rate limit
Enabled: true
Status: running PID:27468 for 1377466s
```

_sms server rate_:
```
SMS server rate
Enabled: true
Status: running PID:27468 for 1377466s
```

_sms server status_:
```
SMS server
Enabled: true
Status: running PID:29157 for 1377474s
```

Jinja template:
```
{
"TEMPLATE-1" : 
	{ 
		"node_types" : [ "node1" ],
		"node_commands" : [ "sms all status", "sms server rate", "sms server status" ],
		"node_regex_commands": ["{{regex_search('Enabled: (.*)'), node_command_output2}}"]
	}
}
```

When the node_commands are executed, we store the output of each into the variable as _node_command_output(X)_
The output of the command _sms server status_ will be stored in variable _node_command_output2_

We want to apply regex to get the value **Enabled** flag
```
{
"TEMPLATE-1" : 
	{ 
		"node_types" : [ "node1" ],
		"node_commands" : [ "sms all status", "ifconfig", "sms server status" ],
		"node_regex_commands": ["{{regex_search('Enabled: (.*)'), node_command_output2}}"]
	}
}
```
Expected output, when we print the _node_regex_commands_:
```
>> print(node_regex_commands)
      true
```

Please let us know if this can this be achieved ?

Thank you!

**Comment by rajeshbadveti** [comment]
We managed to get around it with jinja2 custom filters but would like to hear if there are any other solutions ?

Thanks,
Rajesh Badveti

**Reply by ThiefMaster** [reply]
Custom filters seem to be a good choice for such custom functionality. What other solution are you looking for?

**Reply by rajeshbadveti** [reply]
Thank you!

I came across this URL https://docs.pexip.com/admin/jinja_templates.htm#custom_filters

Question: Why Jinja does not support such inbuilt capability like pex_regex_search('regex pattern', 'string_to_search') ?
